packages/compass-preferences-model/src/preferences.ts

```diff
@@ -0,0 +1,318 @@
+  private _userPreferencesModel: PreferencesAmpersandModel;
+
+  constructor(basepath?: string) {
+    // User preferences are stored to disc via the Ampersand model.
+    const PreferencesModel = Model.extend(storageMixin, {
+      props: preferencesProps,
+      extraProperties: 'ignore',
+      idAttribute: 'id',
+      namespace: 'AppPreferences',
+      storage: {
+        backend: 'disk',
+        basepath,
+      },
+    });
+
+    this._onPreferencesChangedCallbacks = [];
+    this._userPreferencesModel = new PreferencesModel();
+  }
+
+  async fetchPreferences() {
+    const userPreferencesModel = this._userPreferencesModel;
+
+    // Fetch user preferences from the Ampersand model.
+    const fetchUserPreferences = promisifyAmpersandMethod(
+      userPreferencesModel.fetch.bind(userPreferencesModel)
+    );
+
+    try {
+      await fetchUserPreferences();
+    } catch (err) {
+      debug('Failed to load preferences, error while fetching models', {
+        message: (err as Error).message,
+      });
+    }
+
+    return this.getPreferences();
+  }
+
+  async savePreferences(attributes: Partial<GlobalPreferences>) {
+    if (!attributes && isEmpty(attributes)) {
+      return this.getPreferences();
+    }
+
+    const userPreferencesModel = this._userPreferencesModel;
+
+    // Save user preferences to the Ampersand model.
+    const saveUserPreferences: (
+      attributes: Partial<GlobalPreferences>
+    ) => Promise<void> = promisifyAmpersandMethod(
+      userPreferencesModel.save.bind(userPreferencesModel)
+    );
+
+    try {
+      await saveUserPreferences(attributes);
+    } catch (err) {
+      debug('Failed to save preferences, error while saving models', {
+        message: (err as Error).message,
+      });
+    }
+
+    const savedPreferencesValues = this.getPreferences();
+    const changedPreferencesValues = pickBy(
+      savedPreferencesValues,
+      (value, key) => Object.keys(attributes).includes(key)
+    );
+    this._callOnPreferencesChanged(changedPreferencesValues);
+
+    return this.getPreferences();
+  }
+
+  getPreferences() {
+    // TODO: merge user, global, and CLI preferences here.
+    return (
+      this._userPreferencesModel.getAttributes({
+        props: true,
+        derived: true,
+      }) || {}
+    );
+  }
+
+  async getConfigurableUserPreferences() {
+    // Set the defaults and also update showedNetworkOptIn flag.
+    if (!this.getPreferences().showedNetworkOptIn) {
+      await this.savePreferences({
+        autoUpdates: true,
+        enableMaps: true,
+        trackErrors: true,
+        trackUsageStatistics: true,
+        enableFeedbackPanel: true,
+        showedNetworkOptIn: true,
+        theme: THEMES.LIGHT,
+      });
+    }
+    const preferences = this.getPreferences();
+    return Object.fromEntries(
+      Object.entries(preferences).filter(
+        ([key]) => preferencesProps[key].ui === true
+      )
+    );
```

Comment:
```suggestion
    ) as UserPreferences;
```

(always a bit sad that TS can’t really do this for us)

Reply:
Yeeeeah these conversions look not very clean in code.